optimum_benchmark/benchmarks/inference/benchmark.py

```diff
@@ -77,16 +83,33 @@ def run_forward_tracking(self, backend: "Backend") -> None:
         if self.config.memory:
             LOGGER.info("\t+ Tracking forward pass peak memory")
             memory_tracker = memory_tracker_class_for_backend[backend.config.name](backend)
-            with memory_tracker.track(interval=self.config.duration // 100):
+            with memory_tracker.track(interval=self.forward_latency / 10):
                 _ = backend.forward(forward_input)
 
             self.forward_peak_memory = memory_tracker.get_peak_memory()
             LOGGER.info(f"\t+ Forward pass peak memory: {self.forward_peak_memory} (MB)")
 
+        if self.config.energy:
+            LOGGER.info("\t+ Tracking forward pass energy consumption")
+            num_forward_passes = 0
+            energy_tracker = EnergyTracker()
+            with energy_tracker.track(interval=1, file_prefix="forward"):
```

Comment:
With pynvml we cannot get a better frequency than every 1 second?

Reply:
We can get higher measure rates but after experimenting with different values it seems that doesn't change much of the end result. My intuition is that the GPU is already in a steady-state after the warmup and latency tracking (so power doesn't fluctuate a lot) and energy is just `average_power * time`.
Should we use higher frequency nonetheless ?

Reply:
Sounds good, let's keep it that way then :+1: 
Anyway, we'll have opportunities to test various frequencies.